controllers/cloud.redhat.com/providers/providers.go

```diff
@@ -157,16 +157,28 @@ func createResource(cache *rc.ObjectCache, resourceIdent rc.ResourceIdent, nn ty
 	err = cache.Create(resourceIdent, nn, nobj)
 
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error creating object: %s%s %w", nn.Name, nn.Namespace, err)
 	}
 	return nobj, nil
 }
 
 func updateResource(cache *rc.ObjectCache, resourceIdent rc.ResourceIdent, object client.Object) error {
 	err := cache.Update(resourceIdent, object)
```

Comment:
If `err` is `nil` here can we just return? I'm not sure I totally understand but it seems like everything we do after updating the cache is only to contextualize an error we may get. If the cache update succeeds do we really need to do the rest of this stuff?